ddsc/core/ddsapi.py

```diff
@@ -14,32 +15,104 @@ class ContentType(object):
     form = 'application/x-www-form-urlencoded'
 
 
-class KindType(object):
+class DataServiceAuth(object):
     """
-    Holds the types of files supported by DDS.
+    Handles authorization refreshing for DataServiceApi.
     """
-    file_str = 'dds-file'
-    folder_str = 'dds-folder'
-    project_str = 'dds-project'
+    def __init__(self, config):
+        """
+        Setup with initial authorization settings from config.
+        :param config: ddsc.config.Config settings such as auth, user_key, agent_key
+        """
+        self.config = config
+        self._auth = self.config.auth
+        self._expires = None
 
-    @staticmethod
-    def is_file(item):
-        return item.kind == KindType.file_str
+    def get_auth(self):
+        """
+        Gets an active token refreshing it if necessary.
+        :return: str valid active authentication token.
+        """
+        if self.hard_coded_auth():
+            return self._auth
+        if not self.auth_expired():
+            return self._auth
+        self.claim_new_token()
+        return self._auth
 
-    @staticmethod
-    def is_folder(item):
-        return item.kind == KindType.folder_str
+    def claim_new_token(self):
+        """
+        Update internal state to have a new token using a no authorization data service.
+        """
+        # Intentionally doing this manually so we don't have a chicken and egg problem with DataServiceApi.
+        headers = {
+            'Content-Type': ContentType.json,
+        }
+        data = {
+            "agent_key": self.config.agent_key,
+            "user_key": self.config.user_key,
+        }
+        url = self.config.url + "/software_agents/api_token"
+        response = requests.post(url, headers=headers, data=json.dumps(data))
+        if response.status_code == 404:  # server doesn't support user_agent yet
+            msg = "Please add a valid auth token to " + LOCAL_CONFIG_FILENAME + ". Example:\n"
+            msg += "auth: 'eyJ0eXAiOi...YT4Q' "
+            raise ValueError(msg)
+        resp_json = response.json()
+        self._auth = resp_json['api_token']
+        self._expires = resp_json['expires_on']
 
-    @staticmethod
-    def is_project(item):
-        return item.kind == KindType.project_str
+    def hard_coded_auth(self):
```

Comment:
This is the legacy method everyone is using right now.
You hard code a single auth token in your environment or config file and once it's time is up you are out of luck.
Would the `def legacy_auth(self):` be more meaningful.
Or `def non_renewable_auth(self):`

Once the production server goes live with the user auth token this code will be removed.